contracts/staking/BaseShareField.sol

```diff
@@ -16,28 +31,48 @@ contract BaseShareField {
     
     address public shareToken;
     
+    uint public lastRewardBlock;
+    
     struct UserInfo {
         uint amount;     // How many tokens the user has provided.
         uint rewardDebt; // Reward debt. 
         uint rewardEarn; // Reward earn and not minted
+        uint lastRewardTime; // Last time that user got rewards
+        uint multiplierResetTime; // Reset time of multiplier
     }
 
     mapping(address => UserInfo) public users;
-    
+   
+
+    modifier onlyFundManager {
+		require(
+			msg.sender == nameService.getAddress("FUND_MANAGER"),
+			"Only FundManager can call this method"
+		);
+		_;
+	}
+
     function _setShareToken(address _shareToken) internal {
         shareToken = _shareToken;
     }
 
     // Update reward variables of the given pool to be up-to-date.
     function _update() internal virtual {
         if (totalProductivity == 0) {
-            totalShare = totalShare.add(_currentReward());
+            lastRewardBlock = block.number;
             return;
         }
+        FundManager fm = FundManager(nameService.getAddress("FUND_MANAGER"));
+        (uint rewardsPerBlock, uint blockStart, uint blockEnd, bool isBlackListed) = fm.getStakingReward(address(this));
+        if(block.number >= blockStart && blockEnd>= block.number && isBlackListed == false){
+            uint256 multiplier = block.number.sub(lastRewardBlock);
+            uint256 reward = multiplier.mul(rewardsPerBlock * 10 ** 16); // turn it to 18 decimals
+
+            accAmountPerShare = accAmountPerShare.add(reward.mul(1e12).div(totalProductivity));
+            totalShare = totalShare.add(reward);
```

Comment:
what is totalShare? is it like the total rewards this contract has earned  to be divided by  stakers? maybe totalRewards is a better name?

Reply:
No needed in our implementation so I deleted